pkg/events/domain/domain.go

```diff
@@ -50,11 +54,24 @@ type RegisterVendorEvent struct {
 	Keys       []interface{} `json:"keys,omitempty"`
 }
 
-func (r *RegisterVendorEvent) PostProcessUnmarshal() {
+func (r *RegisterVendorEvent) PostProcessUnmarshal(event events.Event) error {
 	// Default fallback to 'healthcare' domain when none is set, for handling legacy data when 'domain' didn't exist.
 	if r.Domain == "" {
 		r.Domain = FallbackDomain
 	}
+	for _, key := range copyKeys(r.Keys) {
```

Comment:
Shouldn't this validation be skipped if no domain exists? Since that would suggest legacy data for which the SubjectAlternativeName doesn't exist